core/src/main/java/io/questdb/log/LogAlertSocketWriter.java

```diff
@@ -0,0 +1,266 @@
+    private HttpLogAlertBuilder alertBuilder;
+    private LogAlertSocket socket;
+    private String alertFooter;
+
+    {
+        if (!alertProps.contains(ORG_ID_ENV)) {
+            alertProps.put(ORG_ID_ENV, DEFAULT_ENV_VALUE);
+        }
+        if (!alertProps.contains(NAMESPACE_ENV)) {
+            alertProps.put(NAMESPACE_ENV, DEFAULT_ENV_VALUE);
+        }
+        if (!alertProps.contains(CLUSTER_ENV)) {
+            alertProps.put(CLUSTER_ENV, DEFAULT_ENV_VALUE);
+        }
+        if (!alertProps.contains(INSTANCE_ENV)) {
+            alertProps.put(INSTANCE_ENV, DEFAULT_ENV_VALUE);
+        }
+        alertProps.put(MESSAGE_ENV, MESSAGE_ENV_VALUE);
+    }
+
+    // changed by introspection
+    private String location = DEFAULT_ALERT_TPT_FILE;
+    private String bufferSize;
+    private String socketAddress;
+
+
+    public LogAlertSocketWriter(RingQueue<LogRecordSink> alertsSrc, SCSequence writeSequence, int level) {
+        this(
+                FilesFacadeImpl.INSTANCE,
+                MicrosecondClockImpl.INSTANCE,
+                alertsSrc,
+                writeSequence,
+                level
+        );
+    }
+
+    public LogAlertSocketWriter(
+            FilesFacade ff,
+            MicrosecondClock clock,
+            RingQueue<LogRecordSink> alertsSrc,
+            SCSequence writeSequence,
+            int level
+    ) {
+        this.ff = ff;
+        this.clock = clock;
+        this.alertsSource = alertsSrc;
+        this.writeSequence = writeSequence;
+        this.level = level & ~(1 << Numbers.msb(LogLevel.ADVISORY)); // switch off ADVISORY
+    }
+
+    @Override
+    public void bindProperties() {
+        int nBufferSize = LogAlertSocket.OUT_BUFFER_SIZE;
+        if (bufferSize != null) {
+            try {
+                nBufferSize = Numbers.parseIntSize(bufferSize);
+            } catch (NumericException e) {
+                throw new LogError("Invalid value for bufferSize");
+            }
+        }
+        socket = new LogAlertSocket(ff, socketAddress, nBufferSize);
+        loadLogAlertTemplate();
+        socket.connect();
+    }
+
+    @Override
+    public void close() {
+        if (socket != null) {
+            socket.close();
+        }
+    }
+
+    @Override
+    public boolean runSerially() {
+        return writeSequence.consumeAll(alertsSource, alertsProcessor);
+    }
+
+    @VisibleForTesting
+    void setBufferSize(String bufferSize) {
+        this.bufferSize = bufferSize;
+    }
+
+    @VisibleForTesting
+    void setLocation(String location) {
+        this.location = location;
+    }
+
+    @VisibleForTesting
+    String getLocation() {
+        return location;
+    }
+
+    @VisibleForTesting
+    void setSocketAddress(String socketAddress) {
+        this.socketAddress = socketAddress;
+    }
+
+    @VisibleForTesting
+    String getSocketAddress() {
+        return socket.getSocketAddress();
```

Comment:
`SocketAddress` should be called `alertTargets`. `SocketAddress` elsewhere in the code is a native structure that is allocated and freed. Here it is a `string`, which makes things unnecessarily confusing